stream-chat-android-client/src/main/java/io/getstream/chat/android/client/setup/state/ClientMutableState.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.client.setup.state
+
+import androidx.annotation.VisibleForTesting
+import io.getstream.chat.android.client.models.ConnectionState
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.setup.state.internal.ClientStateImpl
+import io.getstream.chat.android.core.internal.InternalStreamChatApi
+
+/**
+ * Mutable version of [ClientState]. The class makes possible to change state of the SDK. Should only be used
+ * internally by the SDK.
+ */
+@InternalStreamChatApi
+public interface ClientMutableState : ClientState {
```

Comment:
Why do we need to keep this interface public? Cannot we move state updates to the LLC instead?

Reply:
What do you mean by `move state updates to the LLC`? We need this interface to be public, because its methods are accessed outside of the LLC module. `ClientMutableState.setUser` is called inside, `StreamStatePluginFactory` for example.

Reply:
but do we need to do that in `StreamStatePluginFactory`? LLC passes events to another modules so maybe we can just call `setUser` in LLC?

Reply:
Good point @adasiewiczr, it looks like we could move it to somewhere the `ChatClient` init process happens.

However, I guess @leandroBorgesFerreira didn't want to mess up the way how states were initialized before, to avoid bringing any side-effect issues, cause this PR is quite huge.
So It might be even better to refactor this in the next PR, what do you think?

Reply:
makes sense. Already discussed with @leandroBorgesFerreira